### PR TITLE
Update TF/TFL -> TOSA legalization tests with SSA values associated

### PR DESCRIPTION
**Pull request by armkevincheng**
This is per https://github.com/tensorflow/tensorflow/pull/48193 requested.

Updating existing tests with SSA values to construct a dependency graph.
This is crucial to check complicated lowering like gather, softmax.

### CODE REVIEW

**Comment by armkevincheng** [comment]
Added @stellaraccident and @rsuderman to the reviewer list

**Comment by armkevincheng** [comment]
Previous comment doesn't tag successfully... let me try again.
Add @stellaraccident and @rsuderman to reviewer list.

File: tensorflow/compiler/mlir/tosa/tests/tf-to-tosa-pipeline.mlir [hunk 1]
```diff
@@ -105,13 +105,13 @@ func @test_relu6(%arg0: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
...
-// CHECK: tosa.reshape
-// CHECK: tosa.greater_equal
-// CHECK: tosa.select
+// CHECK-DAG: %[[VAR0:.*]] = "tosa.const"() {value = dense<0.000000e+00> : tensor<f32>}
+// CHECK-DAG: %[[VAR1:.*]] = "tosa.const"() {value = dense<5.000000e-01> : tensor<f32>}
+// CHECK: %[[VAR2:.*]] = "tosa.reshape"(%[[VAR1]]) {new_shape = [1, 1]}
+// CHECK: %[[VAR3:.*]] = "tosa.mul"(%arg0, %[[VAR2]]) {shift = 0 : i32}
+// CHECK: %[[VAR4:.*]] = "tosa.reshape"(%[[VAR0]]) {new_shape = [1, 1]}
+// CHECK: %[[VAR5:.*]] = "tosa.greater_equal"(%arg0, %[[VAR4]])
+// CHECK: %[[VAR6:.*]] = "tosa.select"(%[[VAR5]], %arg0, %[[VAR3]])
```
Comment (rsuderman) [inline tensorflow/compiler/mlir/tosa/tests/tf-to-tosa-pipeline.mlir:114]: Comment regard CHECK vs CHECK-DAG. CHECK-DAG allows different ordering of operations, so it should be preferred in most cases, though the root nodes on the compute graph need to be CHECK statements. In this case, only the "tosa.select" should be a CHECK, the rest should be CHECK-DAGs.

This is needed a many of the checks below.

File: tensorflow/compiler/mlir/tosa/tests/tfl-to-tosa-pipeline.mlir [hunk 2]
```diff
@@ -2,673 +2,662 @@
...
-// CHECK: tosa.const
-// CHECK: tosa.conv2d
-func @test_conv2d(%arg0: tensor<1x32x32x8xf32>, %cst_0: tensor<16x1x1x8xf32>) -> tensor<1x32x32x16xf32> {
+// CHECK-DAG: %[[VAR0:.*]] = "tosa.const"() {value = dense<0.000000e+00> : tensor<16xf32>}
+// CHECK: %[[VAR1:.*]] = "tosa.conv2d"(%arg0, %arg1, %[[VAR0]]) {dilation = [1, 1], pad = [0, 0, 0, 0], stride = [1, 1]}
+func @test_conv2d(%arg0: tensor<1x32x32x8xf32>, %arg1: tensor<16x1x1x8xf32>) -> tensor<1x32x32x16xf32> {
   %cst = constant dense<0.000000e+00> : tensor<16xf32>
-  %0 = "tfl.conv_2d"(%arg0, %cst_0, %cst)  {dilation_h_factor = 1 : i32, dilation_w_factor = 1 : i32, fused_activation_function = "NONE", padding = "SAME", stride_h = 1 : i32, stride_w = 1 : i32}  : (tensor<1x32x32x8xf32>, tensor<16x1x1x8xf32>, tensor<16xf32>) -> tensor<1x32x32x16xf32>
-  return %0 : tensor<1x32x32x16xf32>
+    %0 = "tfl.conv_2d"(%arg0, %arg1, %cst)  {dilation_h_factor = 1 : i32, dilation_w_factor = 1 : i32, fused_activation_function = "NONE", padding = "SAME", stride_h = 1 : i32, stride_w = 1 : i32}  : (tensor<1x32x32x8xf32>, tensor<16x1x1x8xf32>, tensor<16xf32>) -> tensor<1x32x32x16xf32>
```
Comment (rsuderman) [inline tensorflow/compiler/mlir/tosa/tests/tfl-to-tosa-pipeline.mlir:17]: This file has inconsistent indentation throughout. Multiple cases of 8+ spaces before a return statement. Please remove the extra indentation (2 spaces per nesting).

**Review by rsuderman — CHANGES_REQUESTED** [review]
Mostly good, just need some fixes for indentation and CHECK vs CHECK-DAG.

File: tensorflow/compiler/mlir/tosa/tests/tf-to-tosa-pipeline.mlir (hunk 1 above)
Comment (armkevincheng) [reply tensorflow/compiler/mlir/tosa/tests/tf-to-tosa-pipeline.mlir:114]: Thanks. Just apply the fix to all other tests in commit#2

File: tensorflow/compiler/mlir/tosa/tests/tfl-to-tosa-pipeline.mlir (hunk 2 above)
Comment (armkevincheng) [reply tensorflow/compiler/mlir/tosa/tests/tfl-to-tosa-pipeline.mlir]: Fixed in commit#2

**Review by rsuderman — APPROVED** [review]

**Review by rsuderman — CHANGES_REQUESTED** [review]
Imported and tfl-to-tosa-pipeline.mlir appears to be failing.

**Comment by armkevincheng** [comment]
What I'm running now is:
`bazel test --config=v2 tensorflow/compiler/mlir/tosa/... -j 16`
and this runs fine

Is there any way I can reproduce the failure on my side?

**Comment by rsuderman** [comment]
> What I'm running now is:
> `bazel test --config=v2 tensorflow/compiler/mlir/tosa/... -j 16`
> and this runs fine
> 
> Is there any way I can reproduce the failure on my side?

I'm not sure why but bazel on the github repo is not reproducing the issue. I patched in some changes and should be able to land.

For interest sake, it was related to test_matmul not having the correct checks. A few lines of checks were missing.
